StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PaymentSheetVerticalViewController.swift

```diff
@@ -114,6 +153,35 @@ class PaymentSheetVerticalViewController: UIViewController, FlowControllerViewCo
         vc.delegate = self
         bottomSheetController?.pushContentViewController(vc)
     }
+
+    func updateUI() {
+        remove(childViewController: self.paymentMethodListViewController)
+        if let paymentMethodFormViewController = self.paymentMethodFormViewController {
+            remove(childViewController: paymentMethodFormViewController)
+        }
+
+        self.paymentMethodListViewController = VerticalPaymentMethodListViewController(
+            currentSelection: lastVerticalSelection,
+            savedPaymentMethod: selectedPaymentOption?.savedPaymentMethod ?? savedPaymentMethods.first,
+            paymentMethodTypes: paymentMethodTypes,
+            shouldShowApplePay: loadResult.isApplePayEnabled && isFlowController,
+            shouldShowLink: loadResult.isLinkEnabled && isFlowController, // TODO: Edge case where we show Link as button in FC if Apple Pay not enabled
+            rightAccessoryType: rightAccessoryType,
+            appearance: configuration.appearance,
+            delegate: self
+        )
+
+        // If we have only one row in the vertical list and it collects user input, display the form instead of the payment method list.
+        let firstPaymentMethodType = paymentMethodTypes[0]
+        // TODO: Handle offerSaveToLinkWhenSupported, previousCustomerInput, delegate
+        let pmFormVC = PaymentMethodFormViewController(type: firstPaymentMethodType, intent: intent, previousCustomerInput: nil, configuration: configuration, isLinkEnabled: false, delegate: self)
+        if paymentMethodListViewController.rowCount == 1 && pmFormVC.form.collectsUserInput {
+            self.paymentMethodFormViewController = pmFormVC
+            add(childViewController: pmFormVC, containerView: paymentContainerView)
+        } else {
+            add(childViewController: paymentMethodListViewController, containerView: paymentContainerView)
+        }
```

Comment:
Hmm this can't be in here, it should only be called once to determine the initial state. I'll deal with it in a follow-up PR though.